common.go

```diff
@@ -1,4 +1,42 @@
 package sdk
 
+import (
+	"fmt"
+
+	"google.golang.org/grpc"
+)
+
 const NativeBin = "/opt/driver/bin/native"
 const NativeBinTest = "/opt/driver/src/build/native"
+
+const (
+	// MaxMsgSizeCLIName is name of the CLI flag to set max msg size.
+	MaxMsgSizeCLIName = "grpc-max-message-size"
+	// MaxMsgSizeCLIDesc is description for the CLI flag to set max msg size.
+	MaxMsgSizeCLIDesc = "max. message size to send/receive to/from clients (in MB)"
+
+	// DefaulGRPCMaxMsgSizeMb is maximum msg size for gRPC in Mb.
+	DefaulGRPCMaxMsgSizeMb = 100
+	gRPCMaxMsgSizeCapMb    = 2048
+)
+
+//BuildGRPCOptions creats gRPC ServerOption \w maxRecv/Send msg size set.
+// Is intended to be shared by gRPC in bblfshd Server and Drivers.
+// Sets the hard limit of message size to less than 2GB since
+// it may overflow an int value, and it should be big enough.
+func BuildGRPCOptions(maxMessageSizeMb int) ([]grpc.ServerOption, error) {
+	var err error
+	size := maxMessageSizeMb
+	if size >= gRPCMaxMsgSizeCapMb {
+		err = fmt.Errorf("%s=%d is too big (limit is %dMB), using %d instead",
```

Comment:
Do we have a convention about returning errors with non-garbage values? Idiomatically, if err != nil the caller should not use the rest of the response at all. If we do otherwise, we should document that in the function comment. 

I'd suggest we either (a) silently cap the value, and document that we do so; or (b) make the error an actual error.

For (a), 

```
The maximum size limit is capped at X MiB if a larger value is requested.
```

or words to that effect.

Reply:
I see what you mean. But the reason for having this is to get this condition logged, and annoying thing is - logging is only available on the client and we do want to log as this parameter is a frequent source of errors/confusion.

I would leave out propagating a logger (though context or somehow else) for the further refactorings.

So if you prefer strongly only idiomatic use errors, I can revert the logic back and make it un-recoverable condition so the client `os.Exit()` (as it was before).

> If we do otherwise, we should document that in the function comment.
Otherwise, I will go for this option for now.

Reply:
I don't think we should conflate separate concerns: Here, logging and error propagation.  Errors have a semantic meaning, while logging is a monitoring/visibility feature. However, I agree that changing the semantics is probably a larger topic than just this PR, and can be addressed separately. So yeah, let's just document it for now.

I'll only note for future reference that lookout already seems to have a context logging package we might re-use: https://godoc.org/github.com/src-d/lookout/util/ctxlog